src/map.cpp

```diff
@@ -3100,37 +3071,44 @@ bool map::has_nearby_fire( const tripoint &p, int radius )
     return false;
 }
 
-void map::mop_spills( const tripoint &p ) {
+bool map::mop_spills( const tripoint &p )
+{
+    bool retval = false;
     auto items = i_at( p );
     for( auto it = items.begin(); it != items.end(); ) {
```

Comment:
Don't worry about it, it was plenty of information for me to understand the idea. I was just saying that I didn't write it exactly as you did because of those 2 things.

> That means that it only erases one element, not the whole tail.
Yup, so I put it in a loop.